conans/model/editable_cpp_info.py

```diff
@@ -44,22 +56,26 @@ def load(filepath):
         except configparser.Error as e:
             raise ConanException("Error parsing layout file: %s\n%s" % (filepath, str(e)))
         data = OrderedDict()
+        folders = {}
         for section in parser.sections():
             ref, cpp_info_dir = section.rsplit(":", 1) if ':' in section else (None, section)
+            if cpp_info_dir == "folders":
+                folders[ref] = {k: v for k, v in parser.items(section)}
+                continue
             if cpp_info_dir not in EditableCppInfo.cpp_info_dirs:
                 raise ConanException("Wrong cpp_info field '%s' in layout file: %s"
                                      % (cpp_info_dir, filepath))
             if ref:
                 try:
                     r = ConanFileReference.loads(ref)
```

Comment:
It was missing before, but add `validate=True` in the constructor (remember that the default value should be changed to `False` at some moment in the future)